packages/common/src/ipld.ts

```diff
@@ -24,25 +43,101 @@ export const verifyCidForBytes = async (cid: CID, bytes: Uint8Array) => {
   }
 }
 
-export const sha256RawToCid = (hash: Uint8Array): CID => {
+export const sha256ToCid = (hash: Uint8Array, codec: number): CID => {
   const digest = mf.digest.create(sha256.code, hash)
-  return CID.createV1(rawCodec.code, digest)
+  return CID.createV1(codec, digest)
 }
 
-export const cidForCbor = async (data: unknown): Promise<CID> => {
-  const block = await dataToCborBlock(data)
-  return block.cid
+export const sha256RawToCid = (hash: Uint8Array): CID => {
+  return sha256ToCid(hash, rawCodec.code)
 }
 
-export const cborEncode = cborCodec.encode
-export const cborDecode = cborCodec.decode
+export type JsonValue =
+  | boolean
+  | number
+  | string
+  | null
+  | bigint
+  | Array<JsonValue>
+  | { [key: string]: JsonValue }
+  | { [key: number]: JsonValue }
 
-export const cborBytesToRecord = (
-  bytes: Uint8Array,
-): Record<string, unknown> => {
-  const val = cborDecode(bytes)
-  if (!check.is(val, schema.record)) {
-    throw new Error(`Expected object, got: ${val}`)
+export type IpldValue =
+  | boolean
+  | number
+  | string
+  | null
+  | bigint
+  | CID
+  | Uint8Array
+  | Array<IpldValue>
+  | { [key: string]: IpldValue }
+  | { [key: number]: IpldValue }
+
+export const jsonToIpldValue = (val: JsonValue): IpldValue => {
```

Comment:
I was having a bunch of issues with the dag-json library so I just reimplemented 🙃